src/qscan.cpp

```diff
@@ -0,0 +1,874 @@
+		}
+	}
+
+	if (dev->wr_capabilities) {
+		check_write_modes(dev);
+	}
+	if (flags & FL_INFO) {
+		printf(IDEV "Device Generic capabilities : 0x%016LX\n",dev->capabilities);
+		printf(IDEV "Device Read capabilities    : 0x%016LX\n",dev->rd_capabilities);
+		printf(IDEV "Device Write capabilities   : 0x%016LX\n",dev->wr_capabilities);
+	}
+
+	if (dev->wr_capabilities) {
+		printf(IDEV "Supported write modes:");
+		for (int i=0; wr_modes[i].id; i++) {
+			if ((dev->wr_modes & wr_modes[i].id) == wr_modes[i].id) printf(" %s", wr_modes[i].name);
+		}
+		printf("\n");
+	}
+
+
+    if (flags & FL_DINFO) {
+		printf("\nDevice capabilities:\n");	
+		for (int idx=0; capabilities[idx].id || strlen(capabilities[idx].name); idx++) {
+			if (capabilities[idx].id)
+				printf(CAPDEV "%18s: %s\n", capabilities[idx].name,
+					(dev->capabilities & capabilities[idx].id) ? "YES" : "-");
+		}
+
+		if (dev->capabilities & CAP_DVD_CSS) {
+			get_rpc_state(dev);
+			printf(IDEV "  RPC phase       : %d\n", dev->rpc.phase);
+			if (dev->rpc.phase == 2) {
+				if (dev->rpc.region)
+					printf(IDEV "  Cur Region      : %d\n", dev->rpc.region);
+				else
+					printf(IDEV "  Cur Region      : not set\n");
+				printf(IDEV "  Changes left    : %d\n", dev->rpc.ch_u);
+				printf(IDEV "  Resets left     : %d\n", dev->rpc.ch_v);
+			}
+		}
+		printf(CAPDEV "Media is locked   : %s\n", (dev->parms.status & STATUS_LOCK) ? "LOCK": "OFF");
+/*
+		if (get_drive_serial_number(dev)) {
+			text_serial->setText(drive->serial);
+		}
+*/
+		printf("\nVendor-specific features:\n");
+#if 0
+		printf("AudioMaster Q.R.: %s\n", dev->ven_features & YMH_AMQR ? "YES" : "-");
+//		if ((dev->ven_features & YMH_AMQR)       && ((flags & FL_CURRENT) || (flags & FL_YMH_AMQR)))
+//			printf("AudioMaster Q.R.    : %s\n", drive->yamaha.amqr ? "ON":"OFF");
+		printf("Yamaha ForceSpd : %s\n", dev->ven_features & YMH_FORCESPEED ? "YES" : "-");
+//		if ((dev->ven_features & YMH_FORCESPEED)       && ((flags & FL_CURRENT) || (flags & FL_YMH_FORCESPEED)))
+//			printf("Yamaha ForceSpeed   : %s\n", drive->yamaha.forcespeed ? "ON":"OFF");
+#endif
+		printf(CAPVND "Hide CD-R         : %s\n",
+			dev->ven_features & PX_HCDRSS ? (dev->plextor.hcdr ? "ON":"OFF") : "-");
+		printf(CAPVND "SingleSession     : %s\n",
+			dev->ven_features & PX_HCDRSS ? (dev->plextor.sss ? "ON":"OFF") : "-");
+		printf(CAPVND "SpeedRead         : %s\n",
+			dev->ven_features & PX_SPDREAD ? (dev->plextor.spdread ? "ON":"OFF") : "-");
+		printf(CAPVND "Silent mode       : %s\n",
+			dev->ven_features & PX_SILENT ? (dev->plextor_silent.state ? "ON":"OFF") : "-");
+		if ((dev->ven_features & PX_SILENT) && (dev->plextor_silent.state)) plextor_print_silentmode_state(dev);
+		printf(CAPVND "SecuRec           : %s\n",
+			dev->ven_features & PX_SECUREC ? (dev->plextor.securec ? "ON":"OFF") : "-");
+		printf(CAPVND "PoweRec           : %s\n",
+			dev->ven_features & PX_POWEREC ? (dev->plextor.powerec_state ? "ON":"OFF") : "-");
```

Comment:
Codacy found an issue: [Clarify calculation precedence for '&' and '?'.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)